libpod/container_internal_linux.go

```diff
@@ -2367,3 +2378,37 @@ func (c *Container) checkFileExistsInRootfs(file string) (bool, error) {
 	}
 	return true, nil
 }
+
+// getSecret gets a secret to mount into the container
+func (c *Container) getSecret(name string) (string, string, error) {
+	secretsDir := filepath.Join(c.runtime.store.GraphRoot(), "secrets")
+	manager, err := secrets.NewManager(secretsDir)
+	if err != nil {
+		return "", "", err
+	}
+	secr, data, err := manager.LookupSecretData(name)
+	if err != nil {
+		return "", "", err
+	}
+
+	secretFile := filepath.Join(c.config.StaticDir, secr.Name)
+
+	f, err := os.Create(secretFile)
+	if err != nil {
+		return "", "", errors.Wrapf(err, "unable to create %s", secretFile)
+	}
+	defer f.Close()
+	if err := f.Chown(c.RootUID(), c.RootGID()); err != nil {
```

Comment:
Are you sure this should be root? What if the container is run with `--user mheon`, for example?

Reply:
Not sure, I didn't think about that. Is there a UID function I can use for the case of --user?